### PR TITLE
chore: create base repository for file manager

### PR DESCRIPTION
**Pull request by GabyGV-GGV**
## Related Tickets & US
- [CS-60](https://app.clickup.com/t/868b0381t)

## Description

- Continuous integration configuration and sonarqube.
- The project should run without any issues
- Create a new Spring Boot Project for manage unstructured data
It should have the correct dependencies for a NoSQL database or an AWS service.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

This can be tested by running the following commands:

To view dependencies:

`./gradlew build`

To run the main class:

`./gradlew bootRun`

To run tests:

`./gradlew test`

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

- - -
This change is [Reviewable](https://reviewable.io/reviews/SoftDevSix/ArgosFileManager/1)

### CODE REVIEW

**Comment by zrock-dev** [comment]
Dear @gabygabylagaby, please add a description.

File: .github/workflows/check-push.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    steps:
+      - uses: gsactions/commit-message-checker@v2
+        with:
+          pattern: '(?:feat|fix|chore|test|docs|build|ci|style|refactor|perf|revert|hotfix)(?:\(\w+\))?:\s[a-z]{1,2}.+|^Merge branch .+ into .+$'
+          error: 'The commit message must have the following structure: <type>(?): <description>'
+          excludeDescription: 'true'
+          excludeTitle: 'true'
+          checkAllCommitMessages: 'true'
+          accessToken: ${{ secrets.GITHUB_TOKEN }}
```
Comment (teran-joseluis) [inline .github/workflows/check-push.yml]: Don't forget to add the extra empty line.

File: .github/workflows/checkout-pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: deepakputhraya/action-branch-name@master
+        with:
+          regex: '([a-z]+)/([a-z]+)'
+          allowed_prefixes: 'feature,bugfix,hotfix,build,ci,docs,chore,stable,refactor,perf,'
+          ignore: main,develop
+          min_length: 4
+          max_length: 100
```
Comment (teran-joseluis) [inline .github/workflows/checkout-pull-request.yml]: Same here.

File: .github/workflows/checkout-pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+name: CI_Push_Workflow
```
Comment (teran-joseluis) [inline .github/workflows/checkout-pull-request.yml]: Please rename the file according to the action.

File: .github/workflows/sonar-scan.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Generate coverage reports
+        run: ./gradlew jacocoTestReport --no-daemon
+
+      - name: SonarQube Scan
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: ${{ vars.SONAR_HOST_URL }}
+          SONAR_PROJECT_KEY: ${{ vars.SONAR_PROJECT_KEY }}
+        run: ./gradlew sonar --no-daemon
```
Comment (teran-joseluis) [inline .github/workflows/sonar-scan.yml]: Same here about the extra line.

File: .gradle/8.11.1/checksums/checksums.lock
Comment (teran-joseluis) [inline .gradle/8.11.1/checksums/checksums.lock:1]: You shouldn't upload theses files, please a `.gitignore` file for it.

File: .idea/.gitignore
Comment (teran-joseluis) [inline .idea/.gitignore:1]: Same here about the `.gitignore` file. Delete them please.

File: ArgosFileManager/.idea/gradle.xml
Comment (teran-joseluis) [inline ArgosFileManager/.idea/gradle.xml:1]: Already recommendation here.

File: HELP.md
Comment (teran-joseluis) [inline HELP.md:1]: Same recommendation here.

File: build/reports/problems/problems-report.html
Comment (teran-joseluis) [inline build/reports/problems/problems-report.html:1]: Avoid uploading these files please. Use a `.gitignore` please

File: build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(17)
+    }
+}
+
+sonarqube {
+    properties {
+        property("sonar.projectKey", "ArgosFileManager")
+        property("sonar.projectName", "ArgosFileManager")
+        property("sonar.host.url", "http://argos.japaneast.cloudapp.azure.com")
```
Comment (teran-joseluis) [inline build.gradle.kts]: Please use environment variables for this line please and the next one too.

**Review by teran-joseluis — CHANGES_REQUESTED** [review]

File: .github/workflows/check-push.yml (hunk 1 above)
Comment (GabyGV-GGV) [reply .github/workflows/check-push.yml]: Done

File: .github/workflows/checkout-pull-request.yml (hunk 2 above)
Comment (GabyGV-GGV) [reply .github/workflows/checkout-pull-request.yml]: Done

File: .github/workflows/checkout-pull-request.yml (hunk 3 above)
Comment (GabyGV-GGV) [reply .github/workflows/checkout-pull-request.yml]: Done

File: .github/workflows/sonar-scan.yml (hunk 4 above)
Comment (GabyGV-GGV) [reply .github/workflows/sonar-scan.yml]: Done

File: .idea/.gitignore
Comment (GabyGV-GGV) [reply .idea/.gitignore:1]: Done

File: ArgosFileManager/.idea/gradle.xml
Comment (GabyGV-GGV) [reply ArgosFileManager/.idea/gradle.xml:1]: Done

File: HELP.md
Comment (GabyGV-GGV) [reply HELP.md:1]: Done

File: build.gradle.kts (hunk 5 above)
Comment (GabyGV-GGV) [reply build.gradle.kts]: Done

**Review by teran-joseluis — APPROVED** [review]

**Review by zrock-dev — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/SoftDevSix/ArgosFileManager/1)* status: 0 of 40 files reviewed, 11 unresolved discussions (waiting on @gabygabylagaby, @sebastian-barra-jalafund, @teran-joseluis, and @zamoranoalicia)

___
*[`build.gradle.kts` line 22 at r3](https://reviewable.io/reviews/SoftDevSix/ArgosFileManager/1#-OCZxkCB4loov2j-9wqO:-OCZxkCCCzW8IdhgBI4r:bhno8yo) ([raw file](https://github.com/SoftDevSix/ArgosFileManager/blob/543e98046ef7fdec7260a126efa05166148cb96e/build.gradle.kts#L22)):*
> ```Kotlin
>     val sonarHostUrl = System.getenv("SONAR_HOST_URL") ?: ""
>     val sonarToken = System.getenv("SONAR_TOKEN") ?: ""
>     properties {
> ```

Following our convention about dependencies, please update this build file to fetch dependencies from `gradle/libs.versions.toml`. Check the [Argos](https://github.com/SoftDevSix/Argos) as an example.

**Review by sebastian-barra-jalafund — APPROVED** [review]

**Comment by GabyGV-GGV** [comment]
@zrock-dev Done

**Review by zrock-dev — APPROVED** [review]
LGTM

**Pull request merged**